lib/vpago/payway_v2/pre_auth_canceler.rb

```diff
@@ -0,0 +1,27 @@
+require 'faraday'
+
+module Vpago
+  module PaywayV2
+    class PreAuthCanceler < PreAuthCompleter
+
+      # override
+      def hash_data
+        data = "#{merchant_id}#{merchant_auth_encryption}#{req_time}"
+        Base64.strict_encode64(OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha512'), api_key, data))
+      end
+
+      # override
+      def merchant_auth
+        {
+          'mc_id' => merchant_id,
+          'tran_id' => transaction_id
+        }.to_json
+      end
+
+      # override
+      def pre_auth_url
```

Comment:
- let's call it cancel_url instead and use ENV.fetch('PAYWAY_V2_CANCEL_PRE_AUTH_PATH') 
- Add the PAYWAY_V2_CANCEL_PRE_AUTH_PATH to readme and .env.example